### ISSUE TITLE
Error: URL did not return JSON

### ISSUE THREAD

**Issue by i8ramin** [carrot/roots-records#25]
Having some issues using this extension when trying to compile for production. My API endpoint is secure and requires Authorization headers. I've verified that the tokens are correct and the endpoint works when I use Postman.. and does indeed return JSON.

```
  extensions: [
    records(
      locations:
        url:
          path: 'https://www.site.com/api/v1/locations.json'
          headers:
            method: 'GET'
            Authorization: 'Token token=*****************************'
    ),
  ]
```

*Error from the compile:*
```
› roots compile --verbose -e production
compiling...
[Error: URL did not return JSON]
```

*API response in Postman:*
![screen shot 2015-09-17 at 10 30 08 am-1](https://cloud.githubusercontent.com/assets/49030/9935887/4df10316-5d27-11e5-97bc-534c10460a26.png)

Any flags I can pass into roots compile to see more details about the error? Any way I can debug this by adding a breakpoint or something in the node_modules folder for roots? Any help/clues is greatly appreciated.

**Comment by kylemac** [comment]
@i8ramin to be sure, the extensions configuration is from your `app.production.coffee` right?

**Comment by i8ramin** [comment]
@kylemac correct. this is my app.production.coffee file.

**Comment by kylemac** [comment]
@i8ramin i don't have a readily available example that's behind auth to test with, but I know has something to do with this logic: https://github.com/carrot/roots-records/blob/master/lib/index.coffee#L69-L76

if you edit your `node_modules/roots-records/lib/index.js` (lines 108-125 in vanilla-js) you should be able to log the `res` object to get some more info. 

I suspect we're wrongly not passing along the options being passed and therefore the headers. 

Do you mind doing a little debugging?

**Comment by zspecza** [comment]
closed in #26
